### PR TITLE
Welcome Tab Updates

### PR DESCRIPTION
**Pull request by Delphinus8821**
### Description of the Change
The following suggestions in the ticket #943 have been updated and added:
- Only the welcome page launches on start-up of Constellation instead of both the welcome page and tutorial page. The checkbox option for the tutorial page to open on start-up has been moved to the welcome page and the option to display the tutorial page on start-up in the options menu has been converted to the welcome page. The tutorial page has also been updated to mostly contain the "what's new" content with a change to styling for readability.
- The welcome page resize issue has been fixed using a scroll pane.
- The option to start a new graph in selection mode has been added to the welcome page.
- The option to import from JDBC has been added to the welcome page. 

### Alternate Designs
None realised

### Why Should This Be In Core?
These additional options for the welcome page have been added to reflect what has recently been added into Constellation that might be useful for users to have easy access to. 
The tutorial page update to the what's new page reflects what the page is now being used for.

### Possible Drawbacks
None realised

### Applicable Issues
#943

### CODE REVIEW

File: CoreWelcomeView/src/au/gov/asd/tac/constellation/views/welcome/WelcomeViewPane.java [hunk 1]
```diff
@@ -150,7 +158,36 @@ public void handle(ActionEvent e) {
...
+            showOnStartUpCheckBox.setFont(new Font("Arial", 16));
+            lowerLeftHBox.getChildren().add(showOnStartUpCheckBox);
+           
+            showOnStartUpCheckBox.selectedProperty().addListener(new ChangeListener<Boolean>() {
+                @Override
+                public void changed(ObservableValue<? extends Boolean> ov,
+                        Boolean oldVal, Boolean newVal) {
+                    prefs.putBoolean(ApplicationPreferenceKeys.WELCOME_ON_STARTUP, newVal);                     
+                }
+            });
```
Comment (antares1470) [inline CoreWelcomeView/src/au/gov/asd/tac/constellation/views/welcome/WelcomeViewPane.java:177]: should be able to make this a lambda (remove the types of the parameters when you do)

File: CoreWelcomeView/src/au/gov/asd/tac/constellation/views/welcome/WelcomeViewPane.java [hunk 2]
```diff
@@ -150,7 +158,36 @@ public void handle(ActionEvent e) {
...
+            showOnStartUpCheckBox.setSelected(prefs.getBoolean(ApplicationPreferenceKeys.WELCOME_ON_STARTUP, ApplicationPreferenceKeys.WELCOME_ON_STARTUP_DEFAULT));
+            
+            // Create a preferenceListener in order to identify when user preference is changed
+            // Keeps tutorial page and options tutorial selections in-sync when both are open
+            prefs.addPreferenceChangeListener(new PreferenceChangeListener() {
+                @Override
+                public void preferenceChange(PreferenceChangeEvent evt) {
+                    showOnStartUpCheckBox.setSelected(prefs.getBoolean(ApplicationPreferenceKeys.WELCOME_ON_STARTUP, showOnStartUpCheckBox.isSelected()));
+                }
+            });
```
Comment (antares1470) [inline CoreWelcomeView/src/au/gov/asd/tac/constellation/views/welcome/WelcomeViewPane.java:187]: should be able to make this a lambda (remove the type of the parameter when you do)

File: CoreWelcomeView/src/au/gov/asd/tac/constellation/views/welcome/plugins/WhatsNewWelcomePlugin.java [hunk 3]
```diff
@@ -68,7 +68,7 @@ public String getName() {
     @Override
     public void run() {
         SwingUtilities.invokeLater(() -> {
-            final TopComponent tutorial = WindowManager.getDefault().findTopComponent(TutorialTopComponent.class.getSimpleName());
+            final TopComponent tutorial = WindowManager.getDefault().findTopComponent(WhatsNewTopComponent.class.getSimpleName());
```
Comment (antares1470) [inline CoreWelcomeView/src/au/gov/asd/tac/constellation/views/welcome/plugins/WhatsNewWelcomePlugin.java:71]: Should change the name of this variable to reflect the change in class name

File: CorePreferences/src/au/gov/asd/tac/constellation/preferences/Bundle.properties [hunk 4]
```diff
@@ -11,7 +11,7 @@ ApplicationOptionsPanel.autosaveCheckBox.text=Autosave every
 ApplicationOptionsPanel.autosaveLabel.text=minutes
 ApplicationOptionsPanel.startupPanel.AccessibleContext.accessibleName=Startup
 ApplicationOptionsPanel.startupPanel.border.title=Startup
-ApplicationOptionsPanel.startupTutorialCheckbox.text=Display tutorial panel on startup
+ApplicationOptionsPanel.startupTutorialCheckbox.text=Display welcome page on startup
```
Comment (antares1470) [inline CorePreferences/src/au/gov/asd/tac/constellation/preferences/Bundle.properties:14]: should look to rename `startupTutorialCheckbox`

**Review by antares1470 — CHANGES_REQUESTED** [review]
If you're able to update the changelog and whatsnew with update to tutorial page (notably that it isn't the tutorial page anymore) that would be good also

**Review by arcturus2 — COMMENTED** [review]
I wrote this on Monday but forgot to press submit. Some minor UI tweaks if you can look at them @Delphinus8821? If it becomes fiddly to solve don't worry about it - there are more important issues to get your help with.

Screenshot from 2021-01-23 10-06-57

**Comment by Delphinus8821** [comment]
@arcturus2 The scroll bars aren't easily able to be changed. In order for the background colour of the welcome tab to take up the whole window when expanded to full size the pane needs to be set to the size of the screen it's currently on. This will lead to the scroll bars having that excess when shrinking the window.

File: CoreWelcomeView/src/au/gov/asd/tac/constellation/views/welcome/WelcomeViewPane.java [hunk 5]
```diff
@@ -187,23 +228,21 @@ public void handle(ActionEvent e) {
 
                 //Calls the method for the recent graphs to open
                 // on the button action
-                recentGraphButtons[i].setOnAction(new EventHandler<ActionEvent>() {
-                    @Override
-                    public void handle(ActionEvent e) {
-                        RecentFilesWelcomePage.openGraph(text);
-                    }
+                recentGraphButtons[i].setOnAction((ActionEvent e) -> {
```
Comment (antares1470) [inline CoreWelcomeView/src/au/gov/asd/tac/constellation/views/welcome/WelcomeViewPane.java:231]: same here

File: CoreWhatsNewView/nbproject/project.xml [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+                    <run-dependency>
+                        <specification-version>7.53</specification-version>
+                    </run-dependency>
+                </dependency>
+                <dependency>
+                    <code-name-base>org.openide.util</code-name-base>
+                    <build-prerequisite/>
+                    <compile-dependency/>
+                    <run-dependency>
+                        <specification-version>9.15.1</specification-version>
```
Comment (antares1470) [inline CoreWhatsNewView/nbproject/project.xml:259]: Are you able to downgrade this to 9.15? 9.15.1 is only available with netbeans 12.1 (12 is the current LTS)

File: CoreWhatsNewView/nbproject/project.xml [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+                    </run-dependency>
+                </dependency>
+                <dependency>
+                    <code-name-base>org.openide.util.lookup</code-name-base>
+                    <build-prerequisite/>
+                    <compile-dependency/>
+                    <run-dependency>
+                        <specification-version>8.41.1</specification-version>
```
Comment (antares1470) [inline CoreWhatsNewView/nbproject/project.xml:267]: similar here (8.41)

File: CoreWhatsNewView/nbproject/project.xml [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+                    </run-dependency>
+                </dependency>
+                <dependency>
+                    <code-name-base>org.openide.util.ui</code-name-base>
+                    <build-prerequisite/>
+                    <compile-dependency/>
+                    <run-dependency>
+                        <specification-version>9.16.1</specification-version>
```
Comment (antares1470) [inline CoreWhatsNewView/nbproject/project.xml:275]: similar here (9.16)

File: CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/WhatsNewViewPane.java [hunk 9]
```diff
@@ -95,62 +87,22 @@
...
             splitPane.setOrientation(Orientation.HORIZONTAL);
-            tutorialViewPane.setCenter(splitPane);
-
-            //Create left VBox to handle "help" images of menu and mouse
-            VBox leftVBox = new VBox(10);
+            whatsNewViewPane.setCenter(splitPane);
+            
+            //Create left VBox to handle Browser and controls,
+            //or error messages
+            VBox leftVBox = new VBox();
```
Comment (antares1470) [inline CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/WhatsNewViewPane.java:101]: final

File: CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/WhatsNewViewPane.java [hunk 10]
```diff
@@ -187,17 +139,37 @@ public void preferenceChange(PreferenceChangeEvent evt) {
...
                 whatsNewView.getEngine().loadContent(getWhatsNew());
             } catch (ParseException ex) {
                 Exceptions.printStackTrace(ex);
             }
-            rightVBox.getChildren().add(whatsNewView);
+            leftVBox.getChildren().add(whatsNewView);
+
+            
+            //Create left VBox to handle "help" images of menu and mouse
+            VBox rightVBox = new VBox(10);
```
Comment (antares1470) [inline CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/WhatsNewViewPane.java:153]: final

File: CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/WhatsNewViewPane.java [hunk 11]
```diff
@@ -187,17 +139,37 @@ public void preferenceChange(PreferenceChangeEvent evt) {
...
+            splitPane.getItems().add(rightVBox);
+            rightVBox.setPadding(new Insets(10, 10, 10, 10));
+            rightVBox.setAlignment(Pos.TOP_CENTER);
+            rightVBox.setMaxWidth(600);
+            rightVBox.setMinWidth(400);
 
+            //Create images for Left VBox
+            ImageView menuImage = new ImageView(new Image(WhatsNewTopComponent.class.getResourceAsStream(MENU_IMAGE)));
```
Comment (antares1470) [inline CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/WhatsNewViewPane.java:161]: final

File: CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/WhatsNewViewPane.java [hunk 12]
```diff
@@ -187,17 +139,37 @@ public void preferenceChange(PreferenceChangeEvent evt) {
...
+            menuImage.setFitWidth(300);
+            menuImage.setPreserveRatio(true);
+            rightVBox.getChildren().add(menuImage);
+            ImageView mouseImage = new ImageView(new Image(WhatsNewTopComponent.class.getResourceAsStream(MOUSE_IMAGE)));
```
Comment (antares1470) [inline CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/WhatsNewViewPane.java:165]: final

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Comment by cygnus-x-1** [comment]
Love the changes :)

**Review by cygnus-x-1 — APPROVED** [review]

**Review by antares1470 — CHANGES_REQUESTED** [review]
I think some of the constants and fields may be unnecessary in the Welcome Page Pane. It's going to be a bit difficult for me to explain though so I'll create a PR outlining my thoughts.

Also if you're able to resolve the merge conflict that would be great

**Review by antares1470 — APPROVED** [review]
This should be good once the merge conflict is resolved

**Pull request merged**